test/coverage-test/vitest.config.ts

```diff
@@ -43,7 +43,8 @@ export default defineConfig({
     coverage: {
       provider: provider as any,
       customProviderModule: provider === 'custom' ? 'custom-provider' : undefined,
-      include: ['src/**'],
+      include: ['**/src/**', '**/external/**.external.ts'],
+      allowExternal: true,
```

Comment:
Added this for simplicity of adding new tests, but I don't think this is actually testing the allowExternal good enough. I need help with how I would construct these tests.

I have verified that the allowExternal option works by setting include to `['**/.ts', '**/.vue']` and moving the external folder outside the `test/coverage-test` folder:
- if callExternal is calling the function from the external folder:
<img width="586" alt="Screenshot 2023-08-05 at 15 58 21" src="https://github.com/vitest-dev/vitest/assets/20173051/aa9c72f3-7068-4057-a5a0-f710589c7d80">

- if callExternal is not calling the function from the external folder:
<img width="586" alt="Screenshot 2023-08-05 at 15 59 04" src="https://github.com/vitest-dev/vitest/assets/20173051/fe6d1e6c-0c78-42b7-876c-13158cb330b1">


Any ideas how I could write a test like this?

Reply:
Currently the project `root` is `/Users/x/vitest/test/coverage-test`. The `external/` directory is not outside of this directory. Let's remove following files:

- `vitest/test/coverage-test/src/callExternal.ts`
- `vitest/test/coverage-test/external/index.ts`
- `vitest/test/coverage-test/external/file.external.ts`

And add a generic fixture file `vitest/test/test-utils/fixtures/math.ts`:

```ts
export function sum(a: number, b: number) {
  return a + b
}

export function multiply(a: number, b: number) {
  return a * b
}
```

Then in `vitest/test/coverage-test/test/coverage.test.ts` test it as:

```ts
import * as ExternalMath from '../../test-utils/fixtures/math'

test('calling files outside project root', () => {
  expect(ExternalMath.sum(2, 3)).toBe(5)
})
```

And configure `vitest/test/coverage-test/vitest.config.ts` as:

```ts
  include: ['**/src/**', '**/test-utils/fixtures/**'],
  allowExternal: true,
```

This setup will test that `allowExternal: true` works. Our current testing setup doesn't really allow testing different values for different configuration options. We'll need to manually verify that 
`fixtures/math.ts` is not in report when changing value of `allowExternal`.

Reply:
You'll also need to change multiple paths in `coverage-report-tests/*` like:

```diff
- const coveragePath = resolve('./coverage/src')
+ const coveragePath = resolve('./coverage/coverage-test/src')
```

I'm not really sure if we should use `allowExternal: true` for all tests as it changes the setup so much.